Modelica/Blocks/Logical.mo

```diff
@@ -38,20 +52,50 @@ the output is <strong>false</strong>.
 </html>"));
   end Or;
 
-  block Xor "Logical 'xor': y = u1 xor u2"
+  block MultiOr "Logical or of Boolean vector elements"
+    extends Modelica.Blocks.Interfaces.PartialBooleanMISO;
+  equation
+    y=Modelica.Math.BooleanVectors.anyTrue(u);
+    annotation (Icon(coordinateSystem(initialScale=0.1),
+      graphics={ Text(extent={{-90,40},{90,-40}},textString="or")}),
+      Documentation(info="<html>
+<p>
+The output is <strong>true</strong> if any of the elements of the input vector are <strong>true</strong>, otherwise
+the output is <strong>false</strong>.
+</p>
+</html>"));
+  end MultiOr;
+
+  block Nand "Logical 'nand': y = not (u1 and u2)"
     extends Blocks.Interfaces.partialBooleanSI2SO;
   equation
-    y = not ((u1 and u2) or (not u1 and not u2));
+    y = not (u1 and u2);
     annotation (Icon(coordinateSystem(preserveAspectRatio=true, extent={{-100,-100},
               {100,100}}), graphics={Text(
             extent={{-90,40},{90,-40}},
-            textString="xor")}), Documentation(info="<html>
+            textString="nand")}), Documentation(info="<html>
 <p>
-The output is <strong>true</strong> if exactly one input is <strong>true</strong>, otherwise
+The output is <strong>true</strong> if at least one input is <strong>false</strong>, otherwise
 the output is <strong>false</strong>.
 </p>
 </html>"));
-  end Xor;
+  end Nand;
+
+  block MultiNand "Logical nand of Boolean vector elements"
+    extends Modelica.Blocks.Interfaces.PartialBooleanMISO;
+  equation
+    y=not Modelica.Math.BooleanVectors.allTrue(u);
+    annotation (Icon(coordinateSystem(initialScale=0.1), graphics={
+                 Text(extent={{-80,30},{60,-30}},
+            textString="nand",
+            textColor={0,0,0})}),
+      Documentation(info="<html>
+<p>
+The output is <strong>true</strong> if at least one element of the input vector is <strong>false</strong>, otherwise
```

Comment:
This is where we need to deal with the use of `allTrue`:
```suggestion
The output is <strong>true</strong> if the input vector is empty or at least one element of the input vector is <strong>false</strong>, otherwise
```